### ISSUE TITLE
Issue building

### ISSUE THREAD

**Issue by Powersource** [bollu/equinox#1]
I cloned the project, ran `tup init` in the top dir (after make complained) and then `make all` in the bin dir. Here's the specific output https://bpaste.net/show/6af2df502162 I'm very new to both Tup and Rust but I'd probably blame Rust for breaking here.

And I just want to add, I was watching that Carmack video, thought "hey he really should try rust", googled him + rust and got here, where you're linking back to that video :)

**Comment by Powersource** [comment]
[This seems to be the problem](https://github.com/rust-lang/rfcs/blob/master/text/0059-remove-tilde.md) (although I'd be surprised if this is the only deprecated thing we run into). Atm I don't really understand its functionality but the new Box is [documented here](https://doc.rust-lang.org/book/box-syntax-and-patterns.html).

**Comment by bollu** [comment]
Hey! I'd turned off notifications on this repo before :) it would be cool to bring this up again! Do you want to collaborate on this?

**Comment by Powersource** [comment]
Haha, I was just yesterday thinking about mailing you. I'd love to keep trying to get it to work :) A lot of the changes are from when I was a lot worse at rust so we'll see how much I managed to break even more. I'll try to get some commits in this weekend.

**Comment by Powersource** [comment]
If we're gonna use this as a temporary tracking issue, I wanna link the gigantic PR [I just merged here](https://github.com/bollu/equinox/pull/2).

Since mentioning the issue with sfml there they've uploaded a new version to cargo, we should try that one and see how it goes.

**Comment by Powersource** [comment]
@bollu could you enable travis for this repo? The travis.yml is already in here so it should be enough to click some buttons over at travis.

**Comment by bollu** [comment]
Sure thing

**Comment by bollu** [comment]
Yep, I just enabled it. You may need a spurious push to have it run

**Comment by bollu** [comment]
Thanks a ton for all work you're putting in :) is there any way I can help right now? xD

**Comment by Powersource** [comment]
Thanks and no problem, I just want this to compile again :D  
And right now it's progressing fine, but I'll let you know if I get stuck.

**Comment by bollu** [comment]
Sure. Do you want to chat about the design or something sometime? What you're building, etc? I'm curious :)

**Comment by Powersource** [comment]
I haven't thought too much about what to do after it works :D I guess I just want to see how it feels to program a game this way, see how well it works.
